### PR TITLE
[PI-505] Create EPR Product

### PR DESCRIPTION
**Pull request by jaklinger**
## Summary

**Remove items from this list if they are not relevant. Remove this line once this has been done**

- Routine Change
- :exclamation: Breaking Change
- :robot: Operational or Infrastructure Change
- :sparkles: New Feature
- :warning: Potential issues that might be caused by this change

### Details

Add any summary information of what is in the change. **Remove this line if you have nothing to add.**

## Pull Request Naming

Pull requests should be named using the following format:

```text
Tag: [AEA-NNNN] - Short description
```

Tag can be one of:

- `Fix` - for a bug fix. (Patch release)
- `Update` - either for a backwards-compatible enhancement or for a rule change that adds reported problems. (Patch release)
- `New` - implemented a new feature. (Minor release)
- `Breaking` - for a backwards-incompatible enhancement or feature. (Major release)
- `Docs` - changes to documentation only. (Patch release)
- `Build` - changes to build process only. (No release)
- `Upgrade` - for a dependency upgrade. (Patch release)
- `Chore` - for refactoring, adding tests, etc. (anything that isn't user-facing). (Patch release)

If the current release is x.y.z then
- a patch release increases z by 1
- a minor release increases y by 1
- a major release increases x by 1

Correct tagging is necessary for our automated versioning and release process.

The description of your pull request will be used as the commit message for the merge, and also be included in the changelog. Please ensure that your title is sufficiently descriptive.

### Rerunning Checks

If you need to rename your pull request, you can restart the checks by either:

- Closing and reopening the pull request
- pushing an empty commit 
  ```bash
  git commit --allow-empty -m 'trigger build'
  git push
  ```
- Amend your last commit and force push to the branch
  ```bash
  git commit --amend --no-edit
  git push --force
  ```

Rerunning the checks from within the pull request will not use the updated title.

### CODE REVIEW

**Comment by jameslinnell** [comment]
needs some feature tests.

Update the "createCpmProduct.success.feature" and "createCpmProduct.failure.feature", I guess rather than creating new ones.

File: src/layers/domain/repository/cpm_product_repository/v3.py [hunk 1]
```diff
@@ -42,8 +141,7 @@ def read(self, product_team_id, product_id) -> CpmProduct:
         result = self.client.query(**args)
         items = [unmarshall(i) for i in result["Items"]]
         if len(items) == 0:
-            raise ItemNotFound(key=product_id)
-
+            raise ItemNotFound(key=f"{product_team_id}:{product_id}")
```
Comment (megan-bower4) [inline src/layers/domain/repository/cpm_product_repository/v3.py:144]: Do we want the key to be made up of both product team id and product id here? By this stage it should have already been verified that the product team exists? 
(And this isn't the pk for products)

File: src/layers/domain/repository/keys/v3.py [hunk 2]
```diff
@@ -7,3 +7,4 @@ class TableKey(TableKeyAction, StrEnum):
     PRODUCT_TEAM = "PT"
     CPM_SYSTEM_ID = "CSI"
     CPM_PRODUCT = "P"
+    CPM_PRODUCT_KEY = "PK"
```
Comment (megan-bower4) [inline src/layers/domain/repository/keys/v3.py:10]: What are the product keys made up of? (pk and sk)

File: src/layers/domain/repository/cpm_product_repository/v3.py (hunk 1 above)
Comment (megan-bower4) [reply src/layers/domain/repository/cpm_product_repository/v3.py:144]: Agreed this is more helpful - also tailor the error message to specify which object wasn't found, e.g. product team or product

**Review by megan-bower4 — APPROVED** [review]

**Pull request merged**